### PR TITLE
rsync-ssl, testsuite: accept --type=SSL_TYPE anywhere in the args

### PR DESCRIPTION
**Pull request by Zeno-sole**
The --type=... option was only recognized as the first argument, so a command such as "rsync-ssl --dry-run --type=stunnel host::mod" passed the option through to the underlying rsync, which rejected it as unknown.  Scan the full argument list for --type=..., export RSYNC_SSL_TYPE, and drop the option before handing the remaining args to rsync.  The manpage no longer says the option must be first.

Adds a test that runs rsync-ssl with a fake rsync in PATH and checks that --type= is consumed in first, middle, and last positions.

### CODE REVIEW

**Comment by Zeno-sole** [comment]
**Before repair：**
RSYNC_SSL_CA_CERT=/tmp/server.crt rsync-ssl --dry-run --type=stunnel 10.20.8.153::backuprsync: --type=stunnel: unknown option
rsync error: syntax or usage error (code 1) at main.c(1806) [client=3.4.4]

**After repair：**
RSYNC_SSL_CA_CERT=/tmp/server.crt rsync-ssl --dry-run --type=stunnel 10.20.8.153::backup
drwxrwxr-x          4,096 2026/08/13 17:54:16 .
-rw-rw-r--              0 2026/08/13 17:54:16 111

**Comment by Zeno-sole** [comment]
I would like to kindly request a code review from the upstream author/maintainer. @tridge

**Comment by steadytao** [comment]
No need to ping after less then 24 hours. 1) a release has just been made with **major** security concerns, we are resting. 2) its been less then 24 hours :D

Let's be respectful. "kindly" doesn't make something respectful.

File: rsync-ssl [hunk 1]
```diff
@@ -226,9 +226,14 @@ if [[ "$1" == --HELPER ]]; then
...
-fi
+args=()
+for arg in "$@"; do
+    if [[ "$arg" == --type=* ]]; then
+        export RSYNC_SSL_TYPE="${arg#--type=}"
+    else
+        args+=("$arg")
+    fi
+done
+set -- "${args[@]}"
```
Comment (steadytao) [inline rsync-ssl:243]: Stop interpreting wrapper options at --

As written, `rsync-ssl -- --type=stunnel` consumes an operand that was explicitly protected from option parsing. Preserve -- and every subsequent argument, and add a regression for this case.

**Review by steadytao — CHANGES_REQUESTED** [review]

File: rsync-ssl (hunk 1 above)
Comment (Zeno-sole) [reply rsync-ssl:243]: Thanks for the feedback — you're right, and I've fixed it.

  The wrapper-option scan now stops at a -- argument: rsync-ssl preserves the -- and every subsequent argument verbatim, passing them
  through to the underlying rsync unchanged. So a --type=... operand that was explicitly protected from option parsing (e.g. rsync-ssl --
  --type=stunnel host::mod) is no longer consumed by the wrapper, while a --type=... before the -- is still recognized as before.

  I also added regression coverage for both cases in testsuite/rsync-ssl-type-option_test.py (a ---protected operand is preserved, and a
  wrapper option before -- is still consumed), and documented the behavior in rsync-ssl.1.md.

File: testsuite/rsync-ssl-type-option_test.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+base = SCRATCHDIR / 'rsync-ssl-type-opt'
+rmtree(base)
+base.mkdir(parents=True)
+
+args_capture = base / 'rsync_args'
+fakebin = base / 'bin'
+fakebin.mkdir(parents=True)
+fake_rsync = fakebin / 'rsync'
+fake_rsync.write_text(f"#!/usr/bin/env bash\nprintf '%s\\n' \"$@\" > {args_capture}\nexit 0\n")
```
Comment (steadytao) [inline testsuite/rsync-ssl-type-option_test.py]: Please capture and assert `RSYNC_SSL_TYPE` as well as argv. The current test proves that `--type=` is removed but it would also pass if the wrapper silently discarded the requested SSL implementation instead of exporting it. `subprocess.run()` should also check the wrappers exit status.

**Review by steadytao — CHANGES_REQUESTED** [review]

File: testsuite/rsync-ssl-type-option_test.py (hunk 2 above)
Comment (Zeno-sole) [reply testsuite/rsync-ssl-type-option_test.py]: Thanks for the review. This has been addressed in commit `916b75be4d73d8fb9621647353a33a6d9fd3f907`:

1. The fake rsync now captures both the received arguments and `RSYNC_SSL_TYPE`.
2. Each test case asserts the expected `RSYNC_SSL_TYPE`; cases where `--type=` must not be consumed assert that the variable is unset.
3. The test now checks the `rsync-ssl` exit status and reports the exit code and output on failure.

I also verified that the test fails if the wrapper silently discards `--type=`, or if the fake rsync exits with a non-zero status.

**Review by steadytao — APPROVED** [review]
LGTM.
